ipld/read.go

```diff
@@ -2,15 +2,27 @@ package ipld
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/ipfs/go-cid"
 	ipld "github.com/ipfs/go-ipld-format"
 
 	"github.com/celestiaorg/celestia-node/ipld/plugin"
 	"github.com/celestiaorg/nmt"
+
 	"github.com/celestiaorg/nmt/namespace"
 )
 
+type ErrByzantine struct {
+	Index  uint8
+	Shares []*NamespacedShareWithProof
+	IsRow  bool
+}
```

Comment:
Shouldn't we re-use this struct: https://github.com/celestiaorg/rsmt2d/blob/5dc63b1ad762ffe6d65c2a920e9ea49f1fdff327/extendeddatacrossword.go#L31-L36

?

Reply:
Well, we can do this. But we still need custom error to save []nmt.Proof of shares. but instead of one slice.` Shares []*NamespacedShareWithProof` we will have 2:
` Shares []byte`
` Proofs []*nmt.Proof`

Not sure, that this solution is better.